src/dumbo/src/tcp/endpoint.rs

```diff
@@ -522,4 +586,72 @@ mod tests {
             assert_eq!(s.inner().flags_after_ns(), TcpFlags::RST);
         }
     }
+
+    #[test]
+    fn test_parse_request_bytes_error() {
+        // Test unsupported HTTP version.
+        let request_bytes = b"GET http://169.254.169.255/ HTTP/2.0\r\n\r\n";
+        let mut expected_response = Response::new(Version::Http11, StatusCode::NotImplemented);
+        expected_response.set_body(Body::new("Unsupported HTTP version.".to_string()));
+        let actual_response = parse_request_bytes(request_bytes, mock_callback);
+        assert!(actual_response == expected_response);
+
+        // Test invalid HTTP Method.
+        let request_bytes = b"POST http://169.254.169.255/ HTTP/1.0\r\n\r\n";
+        let mut expected_response = Response::new(Version::Http11, StatusCode::NotImplemented);
+        expected_response.set_body(Body::new("Unsupported HTTP method.".to_string()));
+        let actual_response = parse_request_bytes(request_bytes, mock_callback);
+        assert!(actual_response == expected_response);
+
+        // Test invalid HTTP format.
+        let request_bytes = b"GET / HTTP/1.1\r\n";
+        let mut expected_response = Response::new(Version::Http11, StatusCode::BadRequest);
+        expected_response.set_body(Body::new("Invalid request.".to_string()));
+        let actual_response = parse_request_bytes(request_bytes, mock_callback);
+        assert!(actual_response == expected_response);
+    }
+
+    #[test]
+    fn test_unsupported_http_method() {
```

Comment:
This test is not relevant for Dumbo, but it is relevant for MMDS. Can you move it back to MMDS, please? Have in mind also that this test, in MMDS, was checking against `PUT` and `PATCH`, which were not allowed methods, in that context.

Reply:
Thank you for comment. I have moved `test_unsupported_http_method ` from `dumbo` to `mmds` and also added `PUT` and `PATCH` cases.